financial-connections/src/main/java/com/stripe/android/financialconnections/domain/PostAuthorizationSession.kt

```diff
@@ -15,11 +20,39 @@ internal class PostAuthorizationSession @Inject constructor(
 ) {
 
     suspend operator fun invoke(
-        institutionId: String
+        institution: Institution
     ): FinancialConnectionsSessionManifest.FinancialConnectionsAuthorizationSession {
-        return repository.postAuthorizationSession(
-            configuration.financialConnectionsSessionClientSecret,
-            institutionId = institutionId
-        )
+        return try {
+            repository.postAuthorizationSession(
+                configuration.financialConnectionsSessionClientSecret,
+                institutionId = institution.id
+            )
+        } catch (
+            @Suppress("SwallowedException") e: StripeException
+        ) {
+            throw e.toDomainException(institution)
+        }
     }
+
+    private fun StripeException.toDomainException(
+        institution: Institution,
+    ): StripeException = this.stripeError?.let {
+        val institutionUnavailable: String? = it.extraFields?.get("institution_unavailable")
+        val availableAt: String? = it.extraFields?.get("expected_to_be_available_at")
+        when (institutionUnavailable) {
+            "true" -> when {
+                availableAt.isNullOrEmpty() -> InstitutionUnplannedException(
+                    institution = institution,
+                    stripeException = this
+                )
+                else -> InstitutionPlannedException(
+                    institution = institution,
+                    isToday = true,
+                    backUpAt = availableAt.toLong().seconds.inWholeMilliseconds,
+                    stripeException = this
+                )
+            }
+            else -> this
+        }
+    } ?: this
```

Comment:
in order to render the exception, we need information that does not come in the API error response (basically, institution related info). This maps the backend exception into a domain exception that includes this missing information.